kamon-core/src/main/scala/kamon/trace/Span.scala

```diff
@@ -16,155 +16,178 @@
+    private val sampled: Boolean = spanContext.samplingDecision == SamplingDecision.Sample
+    private var operationName: String = initialOperationName
+    private var endTimestampMicros: Long = 0
+
+    private var spanTags = initialTags
+    private var customMetricTags = Map.empty[String, String]
+    private var annotations = List.empty[Span.Annotation]
+
+    def annotate(annotation: Annotation): Span = synchronized {
+      if(sampled && isOpen)
+        annotations = annotation :: annotations
+      this
+    }
 
-  override def setTag(key: String, value: String): Span = synchronized {
-    if (isOpen) {
-      extractMetricTag(key, value)
-      if(sampled)
-        tags = tags ++ Map(key -> value)
+    override def addSpanTag(key: String, value: String): Span = synchronized {
+      if(sampled && isOpen)
+        spanTags = spanTags + (key -> value)
+      this
     }
-    this
-  }
 
-  override def setTag(key: String, value: Boolean): Span = {
-    if (isOpen) {
-      val tagValue = if(value) Span.BooleanTagTrueValue else Span.BooleanTagFalseValue
-      extractMetricTag(key, tagValue)
-      if(sampled)
-        tags = tags + (key -> tagValue)
+    override def addMetricTag(key: String, value: String): Span = synchronized {
+      if(sampled && isOpen && collectMetrics)
+        customMetricTags = customMetricTags + (key -> value)
+      this
     }
-    this
-  }
 
-  override def setTag(key: String, value: Number): Span = {
-    if (isOpen) {
-      val tagValue = String.valueOf(value)
-      extractMetricTag(key, tagValue)
-      if(sampled)
-        tags = tags + (key -> tagValue)
+    override def addBaggage(key: String, value: String): Span = {
+      spanContext.baggage.add(key, value)
+      this
     }
-    this
-  }
 
-  def setMetricTag(key: String, value: String): Span = synchronized {
-    if (isOpen)
-      additionalMetricTags = additionalMetricTags ++ Map(key -> value)
-    this
-  }
+    override def getBaggage(key: String): Option[String] =
+      spanContext.baggage.get(key)
 
-  override def setBaggageItem(key: String, value: String): Span = synchronized {
-    if (isOpen)
-      spanContext.addBaggageItem(key, value)
-    this
-  }
+    override def disableMetricsCollection(): Span = synchronized {
+      collectMetrics = false
+      this
+    }
 
-  override def setOperationName(operationName: String): Span = synchronized {
-    if(isOpen)
-      this.operationName = operationName
-    this
-  }
+    override def context(): SpanContext =
+      spanContext
+
+    override def setOperationName(operationName: String): Span = synchronized {
+      if(isOpen)
+        this.operationName = operationName
+      this
+    }
 
-  private def extractMetricTag(tag: String, value: String): Unit =
-    if(tag.startsWith(Span.MetricTagPrefix))
-      additionalMetricTags = additionalMetricTags ++ Map(tag.substring(Span.MetricTagPrefix.length) -> value)
+    override def finish(): Unit =
```

Comment:
No, the FinishedSpan is not meant to be exposed to the users but to the reporters.